### PR TITLE
Front End Restructure - Offline-first (WIP)

### PR DESCRIPTION
**Pull request by ekrem-aktas**
I've started working on updating the offline-first docs. This PR is WIP

### CODE REVIEW

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md [hunk 1]
```diff
@@ -7,18 +7,34 @@ aliases:
...
 {{% /alert %}}
 
-Mendix supports building offline-first applications for [native mobile](/refguide/native-mobile/) and [progressive web apps](/refguide/mobile/introduction-to-mobile-technologies/progressive-web-app/). Both native and progressive web apps share the same core, and this gives them the same offline-first capabilities. Native mobile apps are always offline-first, but for progressive web apps, it depends on the navigation profile that is configured. The data is stored on the device in a local database, and the files are stored on the file storage of the device.
+Mendix supports building offline-first applications for [native mobile](/refguide/native-mobile/) and [progressive web apps](/refguide/mobile/introduction-to-mobile-technologies/progressive-web-app/). Both native and progressive web apps share the same core, giving them the same offline-first capabilities. Native mobile apps are always offline-first, but for progressive web apps, it depends on the navigation profile that is configured.
```
Comment (jdiehl) [inline content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: I don't understand "but for progressive web apps, it depends on the navigation profile that is configured." What is the configuration option that this depends on and how is it configured?

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md [hunk 2]
```diff
@@ -7,18 +7,34 @@ aliases:
...
+Synchronization is the process of copying data and files from the app's server to the device and updating the server with the changes made on the local device. The local database is populated when the app starts for the first time. After this initial synchronization, data will remain available in the app so that it can work without an internet connection. You can model when and how to perform the subsequent synchronizations. In some cases, the Mendix client may perform a synchronization automatically. Designing an efficient synchronization strategy is crucial for a performant app and a smooth UX. Check out the [Synchronization](/synchronization) page to learn more about it.
+
+## Modelling offline-first applications
+
+You can use the same model concepts (domain model, pages, microflows, etc.) and editors for modeling web applications to model an offline-first application. However, offline-first apps are fundamentally different compared to web apps. Mendix Studio Pro performs validations to ensure your app follows an offline-first approach and works even when there is no connection.
```
Comment (jdiehl) [inline content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: It is valuable to mention that (and what) resources (UI, logic, model) can be shared between offline and online apps, and that a Mendix application can include (and mix) online and offline profiles.

Can you explain the fundamental difference between offline and online apps in a paragraph (maybe put it in a box with a label advanced)? I have observed that our Makers do not understand this difference and that they sometimes design offline and online apps the same way, which leads to an sub-ideal design (e.g. session duration)

Comment (jdiehl) [reply content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: Maybe we should create a sub-page for this topic?

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md [hunk 3]
```diff
@@ -7,18 +7,34 @@ aliases:
...
+
+Visit [this page](best-practices) to learn more about best practices when designing offline-first apps.
+
+## Deploying offline-first applications
```
Comment (jdiehl) [inline content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: This should be consistent with the detail page (Distributed Mobile Apps)

**Comment by jdiehl** [comment]
We should reference all subpages. Where can we add a link to troubleshooting?

**Review by jdiehl — APPROVED** [review]
Looks good to me!

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md (hunk 2 above)
Comment (ConnorLand) [reply content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: As long as @ekrem-aktas can draft a description that's no longer than 3 paragraphs, I think this page would be a good fit for such info.

**Comment by ConnorLand** [comment]
I have added links to all subpages.

> We should reference all subpages. Where can we add a link to troubleshooting?

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md (hunk 3 above)
Comment (ConnorLand) [reply content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: Can you clarify what action I should take here?

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md (hunk 1 above)
Comment (ekrem-aktas) [reply content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: Added a small description and a link to a more detailed document

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md (hunk 2 above)
Comment (ekrem-aktas) [reply content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md]: I tried to expand this section a bit - there may be more differences, but these are the most important ones imo.

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md [hunk 4]
```diff
@@ -17,21 +17,25 @@ Offline-first applications work regardless of the connection to provide a contin
...
+
+Changes made by the user are stored in this offline-first database, too. This means such changes will not exist to other users until the changes are synchronized with the server. When synchronizing the changes made in an app to the server database, there are several aspects to consider for developer teams. For example, the same object may have been edited or removed by a different user, or may no longer be accessible due to access rules defined in the app's model. Agree on workflows at the start of a development project to keep development running smoothly.
+
+### 3.2 Backwards-Compatibility
+
+Another important aspect of developing offline-first apps is backwards-compatibility. Typically when you deploy a new version of a web app to the cloud, all users immediately have access to the latest model. However, that is not the case with offline-first apps. Some parts of your app model are distributed as part of the native mobile app package, such as pages, nanoflows and JavaScript actions. This means even if you change and deploy new versions of these parts, your users do not have access to the latest version until they update their native mobile app package.
```
Comment (ConnorLand) [inline content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md:38]: "This means even if you change and deploy new versions of these parts, your users do not have access to the latest version until they update their native mobile app package." Are these changes correct @ekrem-aktas ? I tried to make the sentence a little clearer and reflect what customers needed to do in order to get a solution (i.e. everything updated correctly).

**Review by ConnorLand — APPROVED** [review]

File: content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md (hunk 4 above)
Comment (ekrem-aktas) [reply content/en/docs/refguide/mobile/using-mobile-capabilities/offlinefirst-data/_index.md:38]: Maybe I can add a bit more explanation about how:

"This means even if you change and deploy new versions of these parts, your users do not have access to the latest version until they update their native mobile apps through an OTA package or AppStore/PlayStore."

**Pull request merged**